src/nb_type.cpp

```diff
@@ -484,7 +486,7 @@ PyObject *nb_type_new(const type_data *t) noexcept {
 
     const char *tp_name = PyUnicode_AsUTF8AndSize(temp_ht->ht_name, nullptr);
 
-    PyObject *result = PyType_GenericAlloc(metaclass, 0);
+    PyObject *result = PyType_GenericAlloc(metaclass, num_members);
```

Comment:
Did you forget to replace `num_members` by `Py_SIZE(temp_tp)` here, or am I missing something?

Reply:
Yeah I guess that's better. Done.